### PR TITLE
CRNS-17: add react-native-navigation to example app and CRNS-149: add multi-file/image upload support

### PR DESCRIPTION
**Pull request by virdesai**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

- CRNS-17: upgrade NativeMessaging app to latest RN version (0.63.2) and added in latest version of react-native-navigation
- CRNS-149: adding support for multi document upload with existing repo but upgraded and multi image upload using react-native-image-crop-picker

### CODE REVIEW

File: native-package/package.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
   },
   "peerDependencies": {
     "@react-native-community/netinfo": ">=2.0.7",
-    "react-native-document-picker": ">=3.2.0",
-    "react-native-image-picker": ">=0.28.1",
+    "react-native-document-picker": ">=3.5.4",
+    "react-native-image-crop-picker": ">=0.33.2",
     "react-native": "*"
```
Comment (vishalnarkhede) [inline native-package/package.json:19]: Lets make it `>= 0.59`

File: native-package/package.json [hunk 2]
```diff
@@ -14,8 +14,8 @@
   },
   "peerDependencies": {
     "@react-native-community/netinfo": ">=2.0.7",
-    "react-native-document-picker": ">=3.2.0",
-    "react-native-image-picker": ">=0.28.1",
+    "react-native-document-picker": ">=3.5.4",
+    "react-native-image-crop-picker": ">=0.33.2",
```
Comment (vishalnarkhede) [inline native-package/package.json:18]: Tutorial will have to be updated as well.

File: native-package/src/index.js [hunk 3]
```diff
@@ -42,34 +42,38 @@ registerNativeHandlers({
...
-        name: res.name,
+        docs: res.map(({ name, uri }) => ({ name, uri })),
+      };
+    } catch (err) {
+      return {
+        cancelled: true,
+      };
+    }
+  },
+  pickImage: async () => {
```
Comment (vishalnarkhede) [inline native-package/src/index.js:61]: We accept `maxNumberOfFiles` prop on MessageInput component. Can we add a param to this function to configure limit?

Comment (vishalnarkhede) [reply native-package/src/index.js:61]: One more thought. We can introduce some prop on MessageInput which enables (or configures) compression on image uploads and pass that prop as argument to this function. Currently we ask users to do it on their end, but instead we can just provide it OOTB (by just adding some prop)

File: native-package/src/index.js [hunk 4]
```diff
@@ -42,34 +42,38 @@ registerNativeHandlers({
...
+      const res = await DocumentPicker.pickMultiple({
         type: [DocumentPicker.types.allFiles],
       });
-      const { uri } = res;
 
       return {
         cancelled: false,
-        uri,
-        name: res.name,
+        docs: res.map(({ name, uri }) => ({ name, uri })),
```
Comment (vishalnarkhede) [inline native-package/src/index.js:53]: :knife:

**Review by vishalnarkhede — CHANGES_REQUESTED** [review]
- [ ] Make sure this issue is fixed - https://github.com/GetStream/stream-chat-react-native/issues/241

- [ ] Works both on iOS and android devices/emulators

- [ ] Update this section of readme (regarding dependency changes) - https://github.com/GetStream/stream-chat-react-native#native-package. 

- [ ] Add extra bullet point about change from react-native-image-picker to react-native-image-crop-picker [here](https://github.com/GetStream/stream-chat-react-native#upgrade)

- [ ] Make sure docs are working fine `yarn docs-server`

File: native-package/src/index.js (hunk 3 above)
Comment (virdesai) [reply native-package/src/index.js:61]: sure

**Comment by vishalnarkhede** [comment]
Should we let all the components go in first as 1.3.0 and then merge this as next one?

**Comment by virdesai** [comment]
> Should we let all the components go in first as 1.3.0 and then merge this as next one?

sure we can wait if you want. it's built off of whatever we have but nbd if we wait

**Comment by jackel119** [comment]
Any estimate as to when this might be merged/released?

**Comment by virdesai** [comment]
> Any estimate as to when this might be merged/released?

@jackel119 waiting on #311 to be merged first and it's almost ready so hopefully today/tomorrow

**Pull request merged**